### ISSUE TITLE
Switching mobile theme does not apply theme

### ISSUE THREAD

**Issue by billreed63** [maqetta/maqetta#1449]
1) Create HTML mobile file
2) Drop Header
3) Switch theme
4)Switch theme again

Old theme is removed but new theme is not applied until page is refreshed

**Comment by jhpedemonte** [comment]
Still seeing this, after reverting the first commit.  I see it on Chrome but not Safari.

Showing one of the split views, though, makes the new theme immediately take effect.

@peller

**Comment by jhpedemonte** [comment]
Also, to see this, you must switch theme by using the **Switch Theme** button in the toolbar.  Then select a new theme in the **Dojo mobile 1.7 theme** drop down.

Switching devices (which also switches themes, if mobile theme set to "device-specific" [the default]) does not exhibit this issue.

**Comment by jhpedemonte** [comment]
The theme also applies if I (1) switch to a different tab in Chrome, (2) go work in a different program for a while (30 secs to 1 min), then (3) go back to Chrome and switch to the Maqetta tab.  At that point, I see the theme take.

**Comment by jhpedemonte** [comment]
I have a theory:

When changing the theme, the code eventually goes into `dojox.mobile.loadDeviceTheme()`, which creates a `<link>` for the new CSS.  However, soon after, this code in **ChangeThemeCommand** reloads the source into the DOM:

``` javascript
        var text = this._context.getModel().getText();
        var e = davinci.Workbench.getOpenEditor();
        e.setContent(e.fileName,text); // force regen of HTML Model to load new theme
```

This is similar to an issue that I, @peller and @billreed63 looked into a while back, where the DOM would get recreated when a CSS load hadn't yet finished.

So I think Bill was probably right in removing the lines that he did in the first commit.  Question then is how do we fix #1524?
